src/Pomelo.EntityFrameworkCore.MySql/Extensions/MySqlDbContextOptionsExtensions.cs

```diff
@@ -87,5 +90,25 @@ private static MySqlOptionsExtension GetOrCreateExtension(DbContextOptionsBuilde
                 ? new MySqlOptionsExtension(existing)
                 : new MySqlOptionsExtension();
         }
+
+        private static bool IsAbleToUseDateTime6(string connectionStr)
```

Comment:
I like the Microsoft suggestion, let me try to implement that.

For the caching, we can use a `ConcurrentDict` keyed off the Connection String, like MySqlConnector does with the [connection pool](https://github.com/mysql-net/MySqlConnector/blob/master/src/MySqlConnector/MySqlClient/ConnectionPool.cs#L118-L131)